cpp/src/arrow/util/byte_stream_split.h

```diff
@@ -545,21 +557,23 @@ void ByteStreamSplitEncodeAvx512(const uint8_t* raw_values, const size_t num_val
 }
 #endif  // ARROW_HAVE_AVX512
 
-#if defined(ARROW_HAVE_SIMD_SPLIT)
+#if defined(ARROW_HAVE_SIMD_DECODE_SPLIT)
 template <typename T>
 void inline ByteStreamSplitDecodeSimd(const uint8_t* data, int64_t num_values,
                                       int64_t stride, T* out) {
 #if defined(ARROW_HAVE_AVX512)
   return ByteStreamSplitDecodeAvx512(data, num_values, stride, out);
 #elif defined(ARROW_HAVE_AVX2)
   return ByteStreamSplitDecodeAvx2(data, num_values, stride, out);
-#elif defined(ARROW_HAVE_SSE4_2)
-  return ByteStreamSplitDecodeSse2(data, num_values, stride, out);
+#elif defined(ARROW_HAVE_SSE4_2) || defined(ARROW_HAVE_NEON)
```

Comment:
Why not `ARROW_HAVE_SIMD_ENCODE_SPLIT`?

Reply:
This branch is included in `#if defined(ARROW_HAVE_SIMD_DECODE_SPLIT)`.
IMO, it seems we shoud separated them by ARCH Macro.